src/app/friends/friends-list/friends-list.component.ts

```diff
@@ -59,18 +69,61 @@ export class FriendsListComponent implements OnInit {
   /** 
    *  Navigate to edit form
    * @param index - number
-   * @returns void
    * */
   navigateToEditForm(index: number): void{
     this.router.navigate(['friends','friends-edit',index]);
   }
 
   /**
-   *  Delete friend
-   * @param index - number
-   * @returns void
+   * Set the friend id in the deleteFriendId for delete the friend
+   * @param name and id of friend for delete
+   */
+  setDeleteFriendId(name:string, id: string): void{
+    this.deleteFriendId = id; 
+    this.deleteFriendName = name;
+  }
+
+  /**
+   *  Delete friend by id
+   * @param id of friend as string
    * */
-  deleteFriend(index: number): void {
-    this.friends.splice(index,1);
+  deleteFriend(id: string): void {
+    this.friendService.deleteFriend(id).subscribe( response => {
+      alert("Delete successfully");
+      this.getFriendList();
+    })
+  }
+
+  /**
+   * Get friend list from the api
+   */
+  getFriendList(): void{
+    this.friendService.getFriendList().subscribe( data => {
+      this.friends = data;
+
+      for(let i=0;i<this.friends.length;i++){
+
+        // Check the friend if they are deleted or not
+        if(this.friends[i].isDeleted === false){
```

Comment:
Optimized way 
if(this.friends[i].isDeleted === false) -> if(!this.friends[i].isDeleted){

Reply:
done